tests/unit/storages/test_request_source_tandem.py

```diff
@@ -0,0 +1,59 @@
+from unittest.mock import create_autospec
+
+from crawlee import Request
+from crawlee.storages import RequestList, RequestQueue, RequestSourceTandem
+from crawlee.storages._request_source import RequestSource
+
+
+async def test_basic_usage() -> None:
+    request_queue = await RequestQueue.open()
+    request_list = RequestList(['http://a.com', 'http://b.com'])
+
+    tandem = RequestSourceTandem(request_list, request_queue)
+    processed = set[str]()
+
+    while not await tandem.is_finished():
+        request = await tandem.fetch_next_request()
+        assert request is not None
+
+        processed.add(request.url)
+        await tandem.add_request(Request.from_url('http://c.com'))
+
+        await tandem.mark_request_as_handled(request)
+
+    assert processed == {
+        'http://a.com',
+        'http://b.com',
+        'http://c.com',
+    }
+
+    assert await tandem.get_handled_count() == 3
+    assert await request_list.get_handled_count() == 2
+    assert await request_queue.get_handled_count() == 3
+
+
+async def test_wait_for_read_only_source() -> None:
```

Comment:
Maybe add docstring about what are we testing here. To me it seems very similar to the previous test and I am not really sure what code change can break this test and not the previous one.